lms/djangoapps/course_wiki/plugins/course_app.py

```diff
@@ -0,0 +1,82 @@
+"""Module with the course app configuration for the Wiki."""
+from typing import Dict, Optional, TYPE_CHECKING
+
+from django.conf import settings
+from django.utils.translation import ugettext_noop as _
+from opaque_keys.edx.keys import CourseKey
+
+from openedx.core.djangoapps.content.course_overviews.models import CourseOverview, CourseOverviewTab
+from openedx.core.djangoapps.course_apps.plugins import CourseApp
+from openedx.core.lib.courses import get_course_by_id
+from xmodule.modulestore.django import modulestore
+from xmodule.tabs import CourseTab, CourseTabList
+
+# Import the User model only for type checking since importing it at runtime
+# will prevent the app from starting since the model is imported before
+# Django's machinery is ready.
+if TYPE_CHECKING:
+    from django.contrib.auth import get_user_model
+
+    User = get_user_model()
+
+WIKI_ENABLED = settings.WIKI_ENABLED
+
+
+class WikiCourseApp(CourseApp):
+    """
+    Course app for the Wiki.
+    """
+
+    app_id = "wiki"
+    name = _("Wiki")
+    description = _("Enable learners to access, and collaborate on course-related information.")
+    documentation_links = {
+        "learn_more_configuration": settings.WIKI_HELP_URL,
+    }
+
+    @classmethod
+    def is_available(cls, course_key: CourseKey) -> bool:  # pylint: disable=unused-argument
+        """
+        Returns if the app is available for the course.
+
+        The wiki is available for all courses or none of them depending on the a Django setting.
+        """
+        return WIKI_ENABLED
+
+    @classmethod
+    def is_enabled(cls, course_key: CourseKey) -> bool:  # pylint: disable=unused-argument
+        """
+        Returns if the wiki is available for the course.
+        """
+        try:
+            wiki_tab = CourseOverview.get_from_id(course_key).tab_set.get(tab_id='wiki')
```

Comment:
Any reason why `CourseTabList` was not used to get the wiki tab here like in L63 (or use `CourseOverview` there)?

Reply:
We don't need to interact with the modulestore here, we can directly fetch this data from the database. 

However, for saving this setting, we need to fetch the course from the modulestore, fetch the actual tab object, change it and save it. Hence, the different approaches.